nvflare/app_common/utils/fl_model_utils.py

```diff
@@ -0,0 +1,123 @@
+    TransferType.MODEL.value: DataKind.WEIGHTS,
+    TransferType.MODEL_DIFF.value: DataKind.WEIGHT_DIFF,
+    TransferType.METRICS_ONLY.value: DataKind.METRICS,
+}
+data_kind_to_transfer_type = {v: k for k, v in transfer_type_to_data_kind.items()}
+
+
+class FLModelUtils:
+    @staticmethod
+    def to_shareable(fl_model: FLModel) -> Shareable:
+        """From FLModel to NVFlare side shareable.
+
+        This is a temporary solution to converts FLModel to the shareable of existing style,
+        so that we can reuse the existing components we have.
+
+        In the future, we should be using the to_dxo, from_dxo directly.
+        And all the components should be changed to accept the standard DXO.
+        """
+        if fl_model.transfer_type.value not in transfer_type_to_data_kind:
+            raise ValueError(f"Invalid TransferType: ({fl_model.transfer_type.value}).")
+
+        data_kind = transfer_type_to_data_kind[fl_model.transfer_type.value]
+
+        dxo = DXO(data_kind, data=fl_model.model, meta={})
+        shareable = dxo.to_shareable()
+        if fl_model.round:
+            shareable.set_header(AppConstants.CURRENT_ROUND, fl_model.round)
+        if fl_model.total_rounds:
+            shareable.set_header(AppConstants.NUM_ROUNDS, fl_model.total_rounds)
+        if fl_model.meta:
+            if AppConstants.VALIDATE_TYPE in fl_model.meta:
+                shareable.set_header(AppConstants.VALIDATE_TYPE, fl_model.meta[AppConstants.VALIDATE_TYPE])
+        return shareable
+
+    @staticmethod
+    def from_shareable(shareable: Shareable) -> FLModel:
+        """From NVFlare side shareable to FLModel.
+
+        This is a temporary solution to converts the shareable of existing style to FLModel,
+        so that we can reuse the existing components we have.
+
+        In the future, we should be using the to_dxo, from_dxo directly.
+        And all the components should be changed to accept the standard DXO.
+        """
+        kwargs = {}
+        dxo = from_shareable(shareable)
+        if dxo.data_kind not in data_kind_to_transfer_type:
+            raise ValueError(f"Invalid shareable with dxo that has data kind: {dxo.data_kind}")
+
+        kwargs[FLModelConst.TRANSFER_TYPE] = TransferType(data_kind_to_transfer_type[dxo.data_kind])
+
+        current_round = shareable.get_header(AppConstants.CURRENT_ROUND, None)
+        total_rounds = shareable.get_header(AppConstants.NUM_ROUNDS, None)
+        validate_type = shareable.get_header(AppConstants.VALIDATE_TYPE, None)
+
+        kwargs[FLModelConst.MODEL] = dxo.data
+        kwargs[FLModelConst.ROUND] = current_round
+        kwargs[FLModelConst.TOTAL_ROUNDS] = total_rounds
+        kwargs[FLModelConst.META] = {AppConstants.VALIDATE_TYPE: validate_type}
+
+        result = FLModel(**kwargs)
+        return result
+
+    @staticmethod
+    def to_dxo(fl_model: FLModel) -> DXO:
+        """Converts FLModel to a DXO."""
+        dxo_dict = {}
```

Comment:
Suggest to rename "dxo_dict" to "attr_dict", because it is not a dict of DXO objects.

Reply:
Thanks Will do